ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/JTATransactionLogXAResourceOrphanFilter.java

```diff
@@ -101,32 +103,24 @@ private boolean transactionLog(Xid xid)
 
         if (!u.equals(Uid.nullUid()))
         {
-            try
-            {
+            if (jtaLogger.logger.isDebugEnabled()) {
+                jtaLogger.logger.debug("Looking for " + u + " and " + transactionType);
+            }
 
+            if (containsCommitMarkableResourceRecord(u) ||
+                    recoveryStore.currentState(u, transactionType) != StateStatus.OS_UNKNOWN)
+            {
                 if (jtaLogger.logger.isDebugEnabled()) {
-                    jtaLogger.logger.debug("Looking for " + u + " and " + transactionType);
+                    jtaLogger.logger.debug("Found record for " + theXid);
                 }
 
-                if (containsCommitMarkableResourceRecord(u) ||
-                        recoveryStore.currentState(u, transactionType) != StateStatus.OS_UNKNOWN)
-                {
-                    if (jtaLogger.logger.isDebugEnabled()) {
-                        jtaLogger.logger.debug("Found record for " + theXid);
-                    }
-
-                    return true;
-                }
-                else
-                {
-                    if (jtaLogger.logger.isDebugEnabled()) {
-                        jtaLogger.logger.debug("No record found for " + theXid);
-                    }
-                }
+                return true;
             }
-            catch (Exception ex)
+            else
             {
-                ex.printStackTrace();
+                if (jtaLogger.logger.isDebugEnabled()) {
+                    jtaLogger.logger.debug("No record found for " + theXid);
+                }
```

Comment:
Looking good. But I do have a quibble, you are attempting to log whether or you find a record but in the exception case no debug output is produced and this would be of interest (unless you know something inside the calls to containsCommitMarkableResourceRecord and recoveryStore is logging the exceptions.